internal/function_time_parser.go

```diff
@@ -518,6 +518,40 @@ func centuryFormatter(t *time.Time) ([]rune, error) {
 	return []rune(fmt.Sprint(t.Year())[:2]), nil
 }
 
+func yearWithoutCenturyParser(text []rune, t *time.Time) (int, error) {
+	const yearLen = 2
+	if len(text) > yearLen {
+		return 0, fmt.Errorf("unexpected year number")
+	}
+	year, err := strconv.ParseInt(string(text), 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("unexpected year number")
+	}
+	if year < 0 {
+		return 0, fmt.Errorf("invalid year number %d", year)
+	}
+	century := 2000
+	if year >= 69 {
+		century = 1900
+	}
+	*t = time.Date(
+		century+int(year),
+		t.Month(),
+		int(t.Day()),
+		int(t.Hour()),
+		int(t.Minute()),
+		int(t.Second()),
+		int(t.Nanosecond()),
+		t.Location(),
+	)
+	return len(text), nil
+}
+
+func yearWithoutCenturyFormatter(t *time.Time) ([]rune, error) {
+	year := t.Format("2006")
```

Comment:
Why is 2006 hard-coded here?

Reply:
@ageiduschek Golang uses a hardcoded datetime for its parse / format string rather than `%X` etc

https://go.dev/src/time/format.go
```
// Here is a summary of the components of a layout string. Each element shows by
// example the formatting of an element of the reference time. Only these values
// are recognized. Text in the layout string that is not recognized as part of
// the reference time is echoed verbatim during Format and expected to appear
// verbatim in the input to Parse.
//
//	Year: "2006" "06"
//	Month: "Jan" "January" "01" "1"
//	Day of the week: "Mon" "Monday"
//	Day of the month: "2" "_2" "02"
//	Day of the year: "__2" "002"
//	Hour: "15" "3" "03" (PM or AM)
//	Minute: "4" "04"
//	Second: "5" "05"
//	AM/PM mark: "PM"
```

`RFC3339     = "2006-01-02T15:04:05Z07:00"`